web/src/main/java/org/cbioportal/genome_nexus/web/mixin/VariantAnnotationMixin.java

```diff
@@ -63,6 +63,10 @@ public Integer getStart()
     @ApiModelProperty(value = "List of transcripts", required = false)
     private List<TranscriptConsequence> transcriptConsequences;
 
+    @JsonProperty(value="successfully_annotated", required = true)
+    @ApiModelProperty(value = "Status flag indicating whether variant was succesfully annotated", required = false)
+    private Boolean successfullyAnnotated;
```

Comment:
It might be a lot of work to change the model more extensively than adding another field, but some thoughts on this:

- how do we store status per resource? Every resource gets a field `succesfully_annotated`?
- where do we store error messages from e.g. VEP and other resources. Is that going in the response as well? It might be nice if there was some filter to enable/disable verbose error reporting
- how do we store errors in our processing of the the raw resources. E.g. `annotation_summary` related errors because some fields are missing. We wouldn't want the whole request to fail, so some catch all errors solution per variant might be good

Anyway adding the field in this manner can be a good start, but just wanted to mention it

Reply:
@inodb good points! Thanks for the feedback. 

- In regards to the first point and third points, what do you mean by resources? Right now we are only adding this successfully annotated flag for variant annotations returned by the annotation controller. The flag just indicates if VEP was able to annotate the passed in variant. I'd be happy to talk about it in more detail to get a more complete picture. 
- This PR is just a first pass for adding the flag. The primary goal was just to ensure we were not silently dropping requested variants. We have discussed expanding it to include error messages but that will have to come down the line. In my picture, we will first need to alter the VEP Wrapper/Runner to include those details in its response.

Reply:
also wanted to mention that it seems like the unit/integration tests behave differently depending on the `application.properties` file sitting in the resources directory (e.g test expects Hgvs format but might run region format). I'm planning on creating an issue for ironing that out.

Reply:
@inodb One idea we had was to add an additional field that stores the exception/error message - what do you think?

Reply:
@averniceday - gotcha! By resources I meant other annotation sources than vep, e.g. my_variant_info, mutation_assessor etc. One tricky thing with adding fields to the model in this way is that if e.g. we want to track success for my_variant_info and we would follow the same design, we would add it there similarly like: my_variant_info.successfully_annotated. But if we want to reuse the my_variant_info model directly by importing it (they use swagger as well), then we might have to transform their model to include that field, which could be complicated. Using more namespacing/nesting in our response could avoid that, because then a particular subtree of the response could be identical to the raw my_variant_info response. But anyway: one step at a time, just something to be aware of

Reply:
@ao508 I think that makes sense. Since we're doing http requests, status codes might be useful as well. But note my point on the nesting. Not sure if that's clear enough I can give an example otherwise

Reply:
To elaborate you can do e.g. like this:

```
annotation: {
    vep: {
      successfully_annotated:
      ensembl_transcript_id:
      ... // other vep related fields
    } 
}
```
or try some other namespace options e.g.:
```
annotation: {
   vep: { // vep fields }
   status: {
       vep: {
           http_code: int
           success: bool
           error_message: string
       }
   }
}
```
Advantage of the latter is that if e.g. VEP would have a OpenAPI spec you could import that easily: 
```
annotation: {
    vep: VEPModel // imported from vep
    status: OurStatusModel
}
```
You might be able to achieve the same by extending the VEPModel like:
```
OurVEPModel extends VEPModel {
     // gets all fields from VEPModel through inheritance plus
     successfullyAnnotated: bool
}
```
But that might break certain things downstream that expect the VEPModel as input (e.g. on the frontend or in a custom script somebody was previously using VEP). As well as that there might be conflicts if we add more status related fields that happen to have the same name (unlikely, but still one can avoid it by design). If we want to stick to changing the existing model `VariantAnnotationMixin` i think namespacing it a bit more might help. E.g.

```
OurVEPModel extends VEPModel {
     // gets all fields from VEPModel through inheritance plus
     annotation_source_status: {
         success: bool
         error_message: string
          // etc
     }
}
```
or something along those lines

Reply:
I agree that when we modify the model defined by VEP / Ensembl we are blurring an externally defined type (which may evolve as the tool / site is updated). So we are vulnerable to clashes and misinterpretations by clients. It would be safer to keep the original type intact/unmodified and to wrap it inside a larger structure.
Also, each time we change to the model delivered by our API we affect clients, requiring rework. So we will be paying additional development costs by an incremental approach where we add a "successfullyAnnotated" flag now, propagate those changes to clients, and then (in a future revision) add a error_message / failure_description field (maybe also wrapping / relocating the status flag), and then propagate *those* changes to clients.
This is one of the detriments of agile programming - you take small incremental steps, but pay the overhead costs of evolving your way to a future solution which is more general / more perfected. Also, there is no guarantee that you ever get there.
If we all agree that it is better to have an embedded, unmodified version of the VEP response model, then we have to choose between keeping what we have now (a tweaked version) and shipping it before reworking it, or reworking it now (delaying our deployment). I'd be in favor of reworking the model to be an embedded model now, but it does expand the scope of what we have planned for this scrum. Opinions @averyniceday @ao508 @mandawilson ? If we want to recommend the embedded unmodified model, we should talk it through with @n1zea144

Reply:
Thanks @sheridancbio! I don't have a strong preference for modifying it now or later.

To add some more context: we have unfortunately already been modifying the VEP model a bit, so maybe it's ok to tweak it further. There are ideas to do a `v2` of this API response. Maybe it's fine that the current one is still a bit hacky as long as we remember the lessons we've learned for the design of a potential new endpoint. Clearly showing in the response what `raw` results are vs `genome nexus` specific is one of those lessons. I have some more ideas about this, but this will be a larger effort.

Good points about the downsides of agile. It often helps you avoid trying to reach perfection, but there can be a lot of overhead with an iterative approach as well.

For this particular PR. I think completely separating the raw response from the status might be a lot of work. A very minimal change though I think is to make the `successfullyAnnotated ` field, a nested object. So instead of:

```
successfully_annotated
```
something like:
```
     annotation_source_status: {
         success: bool
         // TODO: below fields will be implemented later
         // error_message: string
          // etc
     }
```
`annotation_source_status` would then still be a field of the VEPModel but it's nice that all the status related things are grouped already.

That would already be pretty helpful cause you can pass the nested status object around thru various functions more easily on e.g. the frontend and the command line client. When we change that model later on to add more detailed status info (e.g. the error message), then whoever implements those things downstream doesn't have to go looking for which fields are related to status information, because they are all grouped

Reply:
> @averniceday - gotcha! By resources I meant other annotation sources than vep, e.g. my_variant_info, mutation_assessor etc. One tricky thing with adding fields to the model in this way is that if e.g. we want to track success for my_variant_info and we would follow the same design, we would add it there similarly like: my_variant_info.successfully_annotated. But if we want to reuse the my_variant_info model directly by importing it (they use swagger as well), then we might have to transform their model to include that field, which could be complicated. Using more namespacing/nesting in our response could avoid that, because then a particular subtree of the response could be identical to the raw my_variant_info response. But anyway: one step at a time, just something to be aware of

@averyniceday i'm working on integrating Oncokb to annotation, in that pr I import their models and use directly, and it might be complicated to change their model to include that field, not sure is there a good way to indicate the status of these added resources?

Reply:
> @averniceday - gotcha! By resources I meant other annotation sources than vep, e.g. my_variant_info, mutation_assessor etc. One tricky thing with adding fields to the model in this way is that if e.g. we want to track success for my_variant_info and we would follow the same design, we would add it there similarly like: my_variant_info.successfully_annotated. But if we want to reuse the my_variant_info model directly by importing it (they use swagger as well), then we might have to transform their model to include that field, which could be complicated. Using more namespacing/nesting in our response could avoid that, because then a particular subtree of the response could be identical to the raw my_variant_info response. But anyway: one step at a time, just something to be aware of

@averyniceday i'm working on integrating Oncokb to annotation, in that pr I import their models and use directly, and it might be complicated to change their model to include that field, not sure is there a good way to indicate the status of these added resources?

Reply:
@leexgh I'm not familiar with the OncoKB development, do you mean you are also manually interacting with the VariantAnnotation model to set something related to OncoKB? The new code here does something similar. Let me know if you want to talk about it in more detail!